lib/vicadmin/validate.go

```diff
@@ -172,20 +174,42 @@ func NewValidator(ctx context.Context, vch *config.VirtualContainerHostConfigSpe
 
 	// Network Connection Check
 	hosts := []string{
-		"google.com:80",
-		"docker.io:443",
+		"http://google.com",
+		"https://docker.io",
 	}
+
 	nwErrors := []error{}
 
-	for _, host := range hosts {
-		conn, err := net.DialTimeout("tcp", host, time.Minute/2)
+	// create a http client with a custom transport. The transport uses the [s/h]proxy defined in vicmachine create if applicable.
+	client := &http.Client{Timeout: 30 * time.Second}
+	vchArgs := make(map[string]string)
+	for _, args := range vch.VicMachineCreateOptions {
+		splitArgs := strings.Split(args, "=")
+		vchArgs[splitArgs[0]] = splitArgs[1]
+	}
+	//priority given to https proxies
+	proxy := ""
+	if sproxy := vchArgs["--https-proxy"]; sproxy != "" {
```

Comment:
@morris-jason for the change to checking the `ok` portion it is subtle but it does add a difference. In one case we are saying "did the user of this map give us this key with the value `""`"  the case where we use `!ok` protects us from this distinction. We will now be able to tell if someone attempted to give us `""` or did not attempt to populate that field at all.